### PR TITLE
chore(node-builder): display the hardfork info in new line

### PR DESCRIPTION
**Pull request by jsvisa**
Currently the hardfork info looks like below:

image

we can pretty print with `Pre-merge hard forks` as a newline table title:

image

### CODE REVIEW

File: crates/node-builder/src/builder.rs [hunk 1]
```diff
@@ -461,7 +461,7 @@ where
 
         let genesis_hash = init_genesis(provider_factory.clone())?;
 
-        info!(target: "reth::cli", "{}",config.chain.display_hardforks());
+        info!(target: "reth::cli", "\n\n{}", config.chain.display_hardforks());
```
Comment (mattsse) [inline crates/node-builder/src/builder.rs:464]: I'd like to not have empty empty lines
```suggestion
        info!(target: "reth::cli", "\n{}", config.chain.display_hardforks());
```

and ideally we also prevent trailing new line for the last element here

https://github.com/paradigmxyz/reth/blob/e88b16e6e784da7db5fb9cdae3ee7868c3b0790b/crates/primitives/src/chain/spec.rs#L1535-L1537

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic nit

File: crates/node-builder/src/builder.rs (hunk 1 above)
Comment (jsvisa) [reply crates/node-builder/src/builder.rs]: @mattsse could you please give more tips about the tailing newline issue, and just curious why there is a new line after with-merge and post-merge blocks, but no newline after the pre-merge?

Comment (mattsse) [reply crates/node-builder/src/builder.rs]: the bug is here:

https://github.com/paradigmxyz/reth/blob/e88b16e6e784da7db5fb9cdae3ee7868c3b0790b/crates/primitives/src/chain/spec.rs#L1457-L1457

this should just be write!

**Comment by jsvisa** [comment]
@mattsse After the fix, it looks like below:

image

File: crates/primitives/src/chain/spec.rs [hunk 2]
```diff
@@ -1518,23 +1517,36 @@ pub struct DisplayHardforks {
...
-            writeln!(f, "- {fork}")?;
+        fn format(
+            header: &str,
+            forks: &Vec<DisplayFork>,
+            next_is_empty: bool,
+            f: &mut Formatter<'_>,
+        ) -> std::fmt::Result {
+            writeln!(f, "{}:", header)?;
+            for (index, fork) in forks.iter().enumerate() {
+                if next_is_empty && index == forks.len() - 1 {
```
Comment (mattsse) [inline crates/primitives/src/chain/spec.rs:1528]: style: I believe can be solved with:

```rust
let mut iter = forks.iter().enumerate().peekable();
while let Some((index, fork)) = iter.next() {
   write!(f, "- {}", fork)?;
   if iter.peek().is_some() && !next_is_empty {
      writeln!(f)
   }
}
```

**Review by mattsse — CHANGES_REQUESTED** [review]
pedantic style nit

File: crates/primitives/src/chain/spec.rs (hunk 2 above)
Comment (jsvisa) [reply crates/primitives/src/chain/spec.rs]: thanks, this looks more clear and functionable

**Review by mattsse — APPROVED** [review]

**Pull request merged**